Tools/Canvas converter/canvas.py

```
@@ -0,0 +1,212 @@
+    except Exception as e:
+        messagebox.showerror("Error", f"Failed to open image: {e}")
+
+# Function to fetch and display image from a URL
+def fetch_image():
+    url = url_entry.get().strip()
+    try:
+        response = requests.get(url, timeout=10)
+        response.raise_for_status()  # Raise an error if the request fails
+        image = Image.open(BytesIO(response.content))
+        load_image(image)
+    except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)